master/buildbot/master.py

```diff
@@ -325,6 +327,16 @@ def sigusr1(*args):
             self.reactor.stop()
 
         finally:
+            @defer.inlineCallbacks
+            def call_after_signal(sig_num, stack):
+                if not self._got_sigterm:
+                    self._got_sigterm = True
+                    yield self.disownServiceParent()
+                    self.reactor.stop()
+                else:
+                    log.msg('Ignoring SIGTERM, master is already shutting down.')
+
+            signal.signal(signal.SIGTERM, call_after_signal)
```

Comment:
shouldn't we be similar to the code in line 278?

I think the hasattr must be for windows.

Reply:
According to https://docs.python.org/3/library/signal.html#signal.SIGTERM `SIGTERM` is available on Windows unconditionally.

Though your comment got me wondering whether current SIGUSR1 handling is broken on multimaster setups because it is not implemented in terms of `stopService`. In particular, `self.data.updates.masterStopped` will not be called.

Reply:
The issue with masterStopped being not called in clean shutdown may be the cause of https://github.com/buildbot/buildbot/issues/4015.